pkg/operator/observe_config.go

```diff
@@ -219,6 +221,92 @@ func observeCloudProviderNames(listers Listers, existingConfig map[string]interf
 	return observedConfig, errs
 }
 
+func observeClusterCIDRs(listers Listers, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
+	var errs []error
+	clusterCIDRsPath := []string{"extendedArguments", "cluster-cidr"}
+
+	previouslyObservedConfig := map[string]interface{}{}
+	if currentClusterCIDRBlocks, _, _ := unstructured.NestedStringSlice(existingConfig, clusterCIDRsPath...); len(currentClusterCIDRBlocks) > 0 {
+		unstructured.SetNestedStringSlice(previouslyObservedConfig, currentClusterCIDRBlocks, clusterCIDRsPath...)
+	}
+
+	observedConfig := map[string]interface{}{}
+	clusterConfig, err := listers.configmapLister.ConfigMaps("kube-system").Get("cluster-config-v1")
+	if errors.IsNotFound(err) {
+		glog.Warning("configmap/cluster-config-v1.kube-system: not found")
+		return observedConfig, errs
+	}
+	if err != nil {
+		errs = append(errs, err)
+		return previouslyObservedConfig, errs
+	}
+
+	installConfigYaml, ok := clusterConfig.Data["install-config"]
+	if !ok {
+		glog.Warning("configmap/cluster-config-v1.kube-system: install-config not found")
+		return observedConfig, errs
+	}
+	installConfig := map[string]interface{}{}
+	err = yaml.Unmarshal([]byte(installConfigYaml), &installConfig)
+	if err != nil {
+		glog.Warningf("Unable to parse install-config: %s", err)
```

Comment:
why is this just a warning, but the case below is returned as error?

Reply:
:+1: